src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -28,642 +37,878 @@ public class AssemblyLoadTestFixtureCollection : ICollectionFixture<AssemblyLoad
         private AssemblyLoadTestFixtureCollection() { }
     }
 
-    [Collection(AssemblyLoadTestFixtureCollection.Name)]
     public sealed class DefaultAnalyzerAssemblyLoaderTests : TestBase
     {
+#if NETCOREAPP
+        private sealed class InvokeUtil
+        {
+            public void Exec(string typeName, string methodName) => InvokeTestCode(typeName, methodName);
+        }
+#else
+        private sealed class InvokeUtil : MarshalByRefObject
+        {
+            public void Exec(string typeName, string methodName)
+            {
+                try
+                {
+                    InvokeTestCode(typeName, methodName);
+                }
+                catch (TargetInvocationException ex) when (ex.InnerException is XunitException)
+                {
+                    var inner = ex.InnerException;
+                    throw new Exception(inner.Message + inner.StackTrace);
```

Comment:
Is there a tradeoff between throwing this new exception and doing `throw inner` here?

Reply:
Yes. `XunitException` is not serializable so it cannot be thrown across `AppDomain` boundaries. The best I could find is to wrap the content here and rethrow. It's usable enough that I was able to debug all the issues that went through this code path.